Kudu.Services.Web/SafeOnePerRequestHttpModule.cs

```diff
@@ -0,0 +1,34 @@
+﻿namespace Kudu.Services.Web
+{
+    using System.Web;
+    using Ninject.Activation.Caching;
+    using Ninject;
+
+    /// <summary>
+    /// Based on Ninject.Web.Common's OnePerRequestHttpModule
+    /// (https://github.com/ninject/Ninject.Web.Common/blob/14fc90d197f02a528ab81eb2a3f12dccdfacb642/src/Ninject.Web.Common/OnePerRequestHttpModule.cs),
+    /// but does a null check on HttpContext.Current before calling MapKernels. On Linux/Mono, HttpContext.Current is occasionally null
+    /// in the EndRequest stage of the pipeline due to a bug, and the call to MapKernels throws.
+    /// If HttpContext.Current is null, disposal of InRequestScope resources will still happen, it just won't be as aggressive.
```

Comment:
I'll file an issue with them and see what the maintainers say. The null HttpContext is a very weird issue, I've only seen it occur on Mono with our InfoRefsController.